e2e-tests/src/synthetic_network.rs

```diff
@@ -22,3 +33,56 @@ pub async fn set_out_latency(milliseconds: Milliseconds, synthetic_url: String)
         .await
         .expect("unable to commit network configuration");
 }
+
+pub async fn wait_for_further_finalized_blocks(
+    connection: &SignedConnection,
+    blocks_to_wait: u32,
+) -> anyhow::Result<()> {
+    let finalized = connection.get_finalized_block_hash().await?;
+    let finalized_number = connection
+        .get_block_number(finalized)
+        .await?
+        .ok_or(anyhow!(
+            "Failed to retrieve block number for hash {finalized:?}"
+        ))?;
+    let block_number_to_wait = finalized_number + blocks_to_wait;
+    info!(
+        "Current finalized block #{}, waiting for block #{}",
+        finalized_number, block_number_to_wait
+    );
+    connection
+        .wait_for_block(|n| n > block_number_to_wait, BlockStatus::Finalized)
+        .await;
+    Ok(())
+}
+
+pub async fn test_latency_template_test(
```

Comment:
Why is it declared here? Is something else going to use it? It looks specific to some test scenario.

Reply:
Previously `high_out_latency_for_each_quorum` and `high_out_latency_for_all` have duplicated code, this is just a common technique called method extraction.  And wh in this file - it is a helper (utils) file for all latency tests/

Reply:
E.g. please note that in commit https://github.com/Cardinal-Cryptography/aleph-node/pull/1262/commits/a6e03d946dfe4730b4ade1b948da34d78dee5c03 I needed to change template code, whereas in the previous version I'd need to do it twice